Ladybird/Android/src/main/java/org/serenityos/ladybird/WebViewImplementation.kt

```diff
@@ -41,12 +43,18 @@ class WebViewImplementation(
         nativeSetViewportGeometry(nativeInstance, w, h)
     }
 
+    fun setDevicePixelRatio(ratio: Float) {
+        nativeSetDevicePixelRatio(nativeInstance, ratio)
+    }
+
+    // Functions called from native code
     fun bindWebContentService(ipcFd: Int, fdPassingFd: Int) {
-        var connector = WebContentServiceConnection(ipcFd, fdPassingFd)
+        val connector = WebContentServiceConnection(ipcFd, fdPassingFd, resourceDir)
         connector.onDisconnect = {
             // FIXME: Notify impl that service is dead and might need restarted
             Log.e("WebContentView", "WebContent Died! :(")
         }
+        val appContext = parentView.context.applicationContext
```

Comment:
I still don't understand what you need `applicationContext` for, as opposed to the view's own context (which is bound to the activity lifecycle, not that of the app).